src/Processors/Transforms/WindowTransform.cpp

```diff
@@ -1166,6 +1166,23 @@ void WindowTransform::appendChunk(Chunk & chunk)
             // Write out the aggregation results.
             writeOutCurrentRow();
 
+            if (isCancelled())
```

Comment:
It definitely will be devirtualized.

Reply:
Btw note how harder it is to reason about this one-line virtual function compared to just a data field. I had to remember that it is virtual and add `final`. "Object-oriented" features and performance are often at odds...